### ISSUE TITLE
Xgboost4j.java.Booster.evalSet throw java.lang.NumberFormatException when set count:poisson objective

### ISSUE THREAD

**Issue by huaxz1986** [dmlc/xgboost#5582]
When set `objective = reg:linear` , traning is ok. However, when set `objective = count:poisson`, training throw error in spark 0.81:

```
20/04/22 19:13:40 WARN TaskSetManager: Lost task 4.0 in stage 16.0 (TID 20, 100.76.14.183, executor 2): java.lang.NumberFormatException: For input string: "inf"
	at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2043)
	at sun.misc.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
	at java.lang.Float.parseFloat(Float.java:451)
	at java.lang.Float.valueOf(Float.java:416)
	at ml.dmlc.xgboost4j.java.Booster.evalSet(Booster.java:189)
	at ml.dmlc.xgboost4j.java.XGBoost.train(XGBoost.java:194)
	at ml.dmlc.xgboost4j.scala.XGBoost$.train(XGBoost.scala:64)
	at ml.dmlc.xgboost4j.scala.spark.XGBoost$$anonfun$buildDistributedBoosters$1.apply(XGBoost.scala:139)
	at ml.dmlc.xgboost4j.scala.spark.XGBoost$$anonfun$buildDistributedBoosters$1.apply(XGBoost.scala:116)
	at org.apache.spark.rdd.ZippedPartitionsRDD2.compute(ZippedPartitionsRDD.scala:89)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD$$anonfun$7.apply(RDD.scala:337)
	at org.apache.spark.rdd.RDD$$anonfun$7.apply(RDD.scala:335)
	at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:1165)
	at org.apache.spark.storage.BlockManager$$anonfun$doPutIterator$1.apply(BlockManager.scala:1156)
	at org.apache.spark.storage.BlockManager.doPut(BlockManager.scala:1091)
	at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1156)
	at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:882)
	at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:335)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:286)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:123)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1374)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by hcho3** [comment]
Fixed in #5546. Please upgrade to version 1.1.0.

**Comment by AhmedTroudi** [comment]
Hello, I'm using xgboost4j version 1.3.1 and I'm facing the same error when I set `objective=count:poisson`. Can someone help me with this please? Thanks. 
`java.lang.NumberFormatException: For input string: "inf"
[info]  at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2043)
[info]  at sun.misc.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
[info]  at java.lang.Float.parseFloat(Float.java:451)
[info]  at java.lang.Float.valueOf(Float.java:416)
[info]  at ml.dmlc.xgboost4j.java.Booster.evalSet(Booster.java:251)
[info]  at ml.dmlc.xgboost4j.java.XGBoost.trainAndSaveCheckpoint(XGBoost.java:215)
[info]  at ml.dmlc.xgboost4j.java.XGBoost.train(XGBoost.java:284)
[info]  at ml.dmlc.xgboost4j.scala.XGBoost$.$anonfun$trainAndSaveCheckpoint$5(XGBoost.scala:66)
[info]  at scala.Option.getOrElse(Option.scala:189)
[info]  at ml.dmlc.xgboost4j.scala.XGBoost$.trainAndSaveCheckpoint(XGBoost.scala:62)
[info]  at ml.dmlc.xgboost4j.scala.XGBoost$.train(XGBoost.scala:106)
[info]  at ml.dmlc.xgboost4j.scala.spark.XGBoost$.buildDistributedBooster(XGBoost.scala:416)
[info]  at ml.dmlc.xgboost4j.scala.spark.XGBoost$.$anonfun$trainForNonRanking$1(XGBoost.scala:499)
[info]  at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2(RDD.scala:837)
[info]  at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2$adapted(RDD.scala:837)
[info]  at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[info]  at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:349)
[info]  at org.apache.spark.rdd.RDD.$anonfun$getOrCompute$1(RDD.scala:362)
[info]  at org.apache.spark.storage.BlockManager.$anonfun$doPutIterator$1(BlockManager.scala:1371)
[info]  at org.apache.spark.storage.BlockManager.org$apache$spark$storage$BlockManager$$doPut(BlockManager.scala:1298)
[info]  at org.apache.spark.storage.BlockManager.doPutIterator(BlockManager.scala:1362)
[info]  at org.apache.spark.storage.BlockManager.getOrElseUpdate(BlockManager.scala:1186)
[info]  at org.apache.spark.rdd.RDD.getOrCompute(RDD.scala:360)
[info]  at org.apache.spark.rdd.RDD.iterator(RDD.scala:311)
[info]  at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
[info]  at org.apache.spark.scheduler.Task.run(Task.scala:127)
[info]  at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:446)
[info]  at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1377)
[info]  at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:449)
[info]  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info]  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info]  at java.lang.Thread.run(Thread.java:748)
`
